addons/product_images/models/ir_cron_trigger.py

```diff
@@ -0,0 +1,26 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class ir_cron_trigger(models.Model):
+    _inherit = "ir.cron.trigger"
+
+    @api.constrains('cron_id')
+    def _check_cron_id(self):
+        """ Check if `ir_cron_fetch_image` is already triggered
+
+        Ensure that there is a maximum of one trigger at a time for `ir_cron_fetch_image`.
+
+        :return: None
+        :raises ValidationError: If there is already an `ir_cron_fetch_image` triggered
+        """
+        ir_cron_fetch_image_id = self.env.ref('product_images.ir_cron_fetch_image').id
+        if self.cron_id.id != ir_cron_fetch_image_id:
+            return
+        nb_cron = self.env['ir.cron.trigger'].search_count(
+            [('cron_id', '=', ir_cron_fetch_image_id)]
+        )
+        if nb_cron > 1:
+            raise ValidationError(_("You can't trigger this cron more than once at a time!"))
```

Comment:
`UserError` as the user is doing something bad. `ValidationError`s are usually reserved for "inconsistent data" events or similar.

Reply:
I based my code on the documentation, which says constrains should raise a ValidationError (see [odoo.api.constrains](https://www.odoo.com/documentation/14.0/developer/reference/addons/orm.html?highlight=constrains#odoo.api.constrains)). This is a fail-safe for most advanced users to ensure that this cron will never be triggered twice at the same time so I do think that should be considered as an "inconsistent data" event.

Reply:
ValidationError is right here, we are in a constraint